### PR TITLE
Internal error (bug 1180055)

### PR DESCRIPTION
**Pull request by jonasfj**
See bug https://bugzilla.mozilla.org/show_bug.cgi?id=1180055

### CODE REVIEW

**Comment by jonasfj** [comment]
@jhford I recommend that you land PR #46  first... Then this becomes much easier to review.

**Comment by jonasfj** [comment]
I'll rebase this...

**Comment by jonasfj** [comment]
rebased...

File: routes/v1.js [hunk 1]
```diff
@@ -1367,24 +1373,22 @@ api.declare({
     await Promise.all([
       this.queueService.putPendingMessage(task, runId + 1),
       this.publisher.taskPending({
-        status:         status,
+        status,
```
Comment (gregarndt) [inline routes/v1.js]: woah, I didn't realize that you could do this.

File: test/api/resolvetask_test.js [hunk 2]
```diff
@@ -142,16 +142,158 @@ suite('Report task completed', function() {
...
+    assume(s2.runs[0].state).equals('exception');
+    assume(s2.runs[0].reasonResolved).equals('malformed-payload');
+  });
+
+  test("reportException (resource-unavailable) is idempotent", async () => {
+    var taskId = slugid.v4();
+    var allowedToFailNow = false;
+    var gotMessage = null;
+
+    await helper.events.listenFor('except', helper.queueEvents.taskException({
```
Comment (gregarndt) [inline test/api/resolvetask_test.js]: naming the event you're listening for "exception" was just too many characters? :)

File: test/api/resolvetask_test.js [hunk 3]
```diff
@@ -142,16 +142,158 @@ suite('Report task completed', function() {
...
+    });
+
+    debug("### Check status of task (again)");
+    var {status: s2} = await helper.queue.status(taskId);
+    assume(s2.runs[0].state).equals('exception');
+    assume(s2.runs[0].reasonResolved).equals('malformed-payload');
   });
 
-  test("create, claim and reportEception, retry (is idempotent)", async () => {
+  test("reportEception (worker-shutdown) is idempotent", async () => {
```
Comment (gregarndt) [inline test/api/resolvetask_test.js]: "reportException"

File: test/api/resolvetask_test.js (hunk 2 above)
Comment (jonasfj) [inline test/api/resolvetask_test.js]: yeah... it was over 80 chars... so too many...

Luckily tests can be ugly :)

File: test/api/resolvetask_test.js (hunk 3 above)
Comment (jonasfj) [inline test/api/resolvetask_test.js]: Will fix and merge... thanks,

**Pull request merged**